tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -828,6 +885,77 @@ void ordering_test_cases() {
 
         spaceship_test<std::strong_ordering>(c_mem[0], c_mem[0], c_mem[1]);
     }
+    { // optional
+        optional_test<0, 0, 1>();
+        static_assert(optional_test<0, 0, 1>());
+
+#if defined(__cpp_nontype_template_args) && __cpp_nontype_template_args >= 201911
+        optional_test<0.0, 0.0, 1.0>();
+        static_assert(optional_test<0.0, 0.0, 1.0>());
+#endif
+
+        static_assert(
+            std::is_same_v<std::compare_three_way_result_t<std::optional<WeaklyOrdered>, std::optional<WeaklyOrdered>>,
+                std::weak_ordering>);
+        static_assert(std::is_same_v<std::compare_three_way_result_t<std::optional<WeaklyOrdered>, WeaklyOrdered>,
+            std::weak_ordering>);
+        static_assert(std::is_same_v<std::compare_three_way_result_t<std::optional<WeaklyOrdered>, std::nullopt_t>,
+            std::strong_ordering>);
+    }
+    { // tuple
+        tuple_like_test<std::tuple>();
+        static_assert(tuple_like_test<std::tuple>());
+
+        {
+            constexpr std::tuple<int, double> t1{1, 1.0};
+            constexpr std::tuple<double, double> t1_equal{1.0, 1.0};
+            constexpr std::tuple<int, double> t2{2, 1.0};

```

Comment:
These extra tests can be moved into the "dual-mode" `tuple_like_test` to avoid repeating compile-time and runtime (except for `std::tuple<>` of course), although the cross-type tests need to be guarded with `if constexpr` because `pair` annoyingly doesn't support such comparisons.